src/core.js

```diff
@@ -1944,8 +1944,9 @@ function/* 12.1.1.1 */InitializeDateTimeFormat (dateTimeFormat, locales, options
         pattern,
 
     // 31. Let hr12 be the result of calling the GetOption abstract operation with
-    //     arguments options, "hour12", "boolean", undefined, and undefined.
-        hr12 = GetOption(options, 'hour12', 'boolean'/*, undefined, undefined*/);
+    //     arguments options, "hour12", "boolean", undefined, and the internal
+    //     value of "hour12" from the best format match as the result back value.
+        hr12 = GetOption(options, 'hour12', 'boolean', undefined, bestFormat.hour12);
```

Comment:
I'm confused by this part because it's counter to the [specification](http://www.ecma-international.org/ecma-402/1.0/#sec-12.1.1.1).  I'm not sure that we should be deviating in this way and it could introduce other parity issues with browser implementations.

(Note that our steps are out of sync with the spec because steps 20, 21 and 22 should be a, b and c.)